### ISSUE TITLE
Feature Request: Allow Spawning Facets from Public Folders

### ISSUE THREAD

**Issue by code807** [Yellow-Dog-Man/Resonite-Issues#1680]
### Is your feature request related to a problem? Please describe.

Currently facets can only be spawned into userspace after first spawning them into a world and saving them to a personal folder.

### Describe the solution you'd like

Allow the spawning of facets directly from public folders via the Spawn Facet button.

### Describe alternatives you've considered

The current solution is to save every desired facet to a personal folder first.

### Requesters

code807

**Comment by Frooxius** [comment]
We won't be adding this until we have stronger sandboxing/protection mechanisms.

Facets in Userspace exist in privileged space which contains more sensitive information. Therefore we want the spawning of one in userspace to be a **very** deliberate action - therefore requiring it to be re-saved into the inventory.

When it's in any public folder, it's easy to just spawn it without much thinking.

**Comment by 5H4D0W-X** [comment]
Making the "spawn facet" button a click-and-confirm action, much like item deletion, might be a good option here.

**Comment by Frooxius** [comment]
I don't feel that's sufficient. People can double click things absent-mindedly.

It also allows the facet to be updated without user's knowledge. Spawning and saving specific copy locks it to that version.

**Comment by TisFoolish** [comment]
> We won't be adding this until we have stronger sandboxing/protection mechanisms.

Would the same hold true for something like having a context menu option to spawn a grabbed facet in user space? Different feature so it should go into its own request, I'm just curious.

**Comment by Frooxius** [comment]
Yes. That also makes it very quick and easy to bring something potentially dangerous to userspace, which is what we want to avoid.

We want it to be a lot more deliberate action for the time being.

**Comment by TisFoolish** [comment]
I would say having to grab the facet, open the context menu, and then click the spawn button is an extremely deliberate action. BUT I understand that it may not be as deliberate as you'd like.

**Comment by TisFoolish** [comment]
Shouldn't this stay open? Froox didn't say that this isn't planned, just that something like this would first need #1103
